src/lib/middleware.ts

```diff
@@ -1,89 +1,11 @@
 import { CookieSerializeOptions, serialize } from "cookie";
-import { IncomingHttpHeaders, IncomingMessage, ServerResponse } from "http";
-import {
-  GetServerSideProps,
-  GetServerSidePropsContext,
-  GetServerSidePropsResult,
-} from "next";
+import { IncomingMessage, ServerResponse } from "http";
+import { GetServerSidePropsContext } from "next";
 import { NextApiRequestCookies } from "next/dist/next-server/server/api-utils";
 import parse from "urlencoded-body-parser";
 import { v4 as uuidv4 } from "uuid";
-import { FormCacheFactory, FormSubmission, IFormCache } from "./formCache";
-import { Registration } from "./registration/registration";
-import { acceptRejectCookieId, formSubmissionCookieId } from "./types";
-import { toArray } from "./utils";
-
-export type BeaconsContext = GetServerSidePropsContext & {
-  showCookieBanner: boolean;
-  submissionId?: string;
-  formData: FormSubmission;
-  registration?: Registration;
-  useIndex: number;
-};
-
-export function withCookieRedirect<T>(callback: GetServerSideProps<T>) {
-  return async (
-    context: BeaconsContext
-  ): Promise<GetServerSidePropsResult<T>> => {
-    const cookies: NextApiRequestCookies = context.req.cookies;
-
-    if (!cookies || !cookies[formSubmissionCookieId]) {
-      return {
-        redirect: {
-          destination: "/",
-          permanent: false,
-        },
-      };
-    }
-
-    return callback(context);
-  };
-}
```

Comment:
Replaced by the explicit rule `IfUserHasNotStartedEditingADraftRegistration`.  (I'm not wedded to the name).